### PR TITLE
Get Apps card: Iterate on link to jetpack.com/apps to ensure backwards compatibility and click-tracking

### PR DESCRIPTION
**Pull request by SiobhyB**
#### Changes proposed in this Pull Request:

This PR iterators on the link to `jetpack.com/apps` that can be found in the Get App card via _Jetpack_ → _Dashboard_. This includes the following changes:

* The [redirect helper](https://github.com/Automattic/jetpack/tree/trunk/projects/js-packages/components/tools/jp-redirect) is used to build the URL. Note, the instructions at PCYsg-pY7-p2 will be followed to set up the redirect once the naming proposed in this PR has been finalised/approved.
* An `onClick` function -- `trackAppLinkClick` -- has been added to the link to help us to track the times the links been clicked on. Similar to the above, the necessary internal processes to finish setting up the tracking will be done once we've agreed on the naming/code. 
* In addition, `rel="noopener noreferrer"` and `target="_blank"` have been added as values to the link, inline with the way similar links are handled throughout the Jetpack codebase. For example, see [this link](https://github.com/Automattic/jetpack/blob/510f1bf711062b55592ddcee7e5f6bee321c93b4/projects/plugins/jetpack/_inc/client/recommendations/sidebar/mobile-app/index.tsx#L85-L90) to the jetpack.com/mobile.

#### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Jetpack product discussion

N/A

#### Does this pull request change what data or activity we track or use?

Yes, this PR adds tracking for the times a user clicks on a link to the jetpack.com/apps page.

#### Testing instructions:

* Navigate to _Jetpack_ → _Dashboard_ to ensure this change doesn't alter the appearance of the apps card in any way.
* Note, the link won't functionally work until a redirect has been set up, which will happen once the naming in this PR has been finalised/approved.

Screenshot 2022-10-06 at 15 24 07

### CODE REVIEW

File: projects/plugins/jetpack/_inc/client/components/apps-card/index.jsx [hunk 1]
```diff
@@ -71,7 +79,15 @@ class AppsCard extends React.Component {
 						'jetpack'
 					),
 					{
-						a: <a className="jp-apps-card__link" href="https://jetpack.com/app" />,
+						a: (
+							<a
+								className="jp-apps-card__link"
+								href={ getRedirectUrl( 'jetpack-plugin-dashboard-apps-card' ) }
```
Comment (SiobhyB) [inline projects/plugins/jetpack/_inc/client/components/apps-card/index.jsx:85]: Note, this redirect hasn't yet been activated in the backend. I wanted to await feedback first. Once the PR is approved and we're agreed on naming, I'll follow the steps to add the redirect from here: PCYsg-pY7-p2

File: projects/plugins/jetpack/_inc/client/components/apps-card/index.jsx [hunk 2]
```diff
@@ -24,6 +25,13 @@ class AppsCard extends React.Component {
 		} );
 	};
 
+	trackAppLinkClick = () => {
+		analytics.tracks.recordJetpackClick( {
+			target: 'jetpack-apps-link',
```
Comment (SiobhyB) [inline projects/plugins/jetpack/_inc/client/components/apps-card/index.jsx:30]: Similar to [my note around the redirect](https://github.com/Automattic/jetpack/pull/26668/files#r989068201), I'll work to make the internal changes necessary for this to work once this code's been approved.

Comment (fluiddot) [reply projects/plugins/jetpack/_inc/client/components/apps-card/index.jsx:30]: I noticed that `recordJetpackClick` uses a generic event named `jetpack_wpa_click` ([reference](https://github.com/Automattic/jetpack/blob/966b9aaf4583ed2407b0c26cf14d05f09185bf4f/projects/js-packages/analytics/index.jsx#L165)) so we won't need to register the event internally in Tracks.

**Review by fluiddot — APPROVED** [review]
LGTM 🎊 !

**NOTE:** As mentioned in [this comment](https://github.com/Automattic/jetpack/pull/26668/files#r989068201), clicking in the  `jetpack.com/app` link returns a 404 because the redirection is not registered. **@SiobhyB  Before merging this PR we should ensure that the link works.**

File: projects/plugins/jetpack/_inc/client/components/apps-card/index.jsx (hunk 1 above)
Comment (SiobhyB) [reply projects/plugins/jetpack/_inc/client/components/apps-card/index.jsx:85]: I've gone ahead to set this up now. :)

File: projects/plugins/jetpack/_inc/client/components/apps-card/index.jsx (hunk 2 above)
Comment (SiobhyB) [reply projects/plugins/jetpack/_inc/client/components/apps-card/index.jsx:30]: Ah, perfect, thank you for digging into that Carlos!

**Comment by SiobhyB** [comment]
Thank you for reviewing @fluiddot! 🙇‍♀️ I've set up the redirect now and have gone ahead to set this PR as ready for a final review by the crew.

**Review by coder-karen — APPROVED** [review]
Works well, I can follow the jetpack_wpa_click event in the tracks live view as well.

**Pull request merged**